### ISSUE TITLE
Getting started / example

### ISSUE THREAD

**Issue by roberth** [fluidattacks/makes#511]
Hi,
This project looks really interesting. Do you have an example repo that readers can use as a starting point for experimentation?

**Comment by blaggacao** [comment]
https://gitlab.com/fluidattacks/product/-/tree/master/makes (not precisely entry-level though) :smiley:

**Comment by dsalaza4** [comment]
As of today we do not have something such as an example repository.

I would advise:

1. Reading the [Reference section](https://github.com/fluidattacks/makes#makesnix-reference) for using builtins.
2. Reading the [Extending Makes section](https://github.com/fluidattacks/makes#extending-makes) for adding your own functionalities/jobs.
3. Reading [makes.nix](https://github.com/fluidattacks/makes/blob/main/makes.nix) as it provides a working baseline for using some builtins.

**Comment by kamadorueda** [comment]
I also understand how important it is for having extensive documentation for new people. Sometimes we (devs) forget about the adoption and marketing part,

So right now what Daniel said is what we have, but I'll try to find the time to improve the docs. We also accept feedback (issues) and contributions on the rough edges

Thanks for pointing this out!

**Comment by ldeck** [comment]
> Hi,
> This project looks really interesting. Do you have an example repo that readers can use as a starting point for experimentation?

Yes @kamadorueda I agree the project looks really promising for use. The additions to the README that I would suggest include the following:
1. **Getting Started > Installation: Per Project**
  a. Using nix mkShell and direnv. 
    Installation of makes globally is not (or should not imho) be a recommended pattern for developers managing numerous projects. The following snippet demonstrates how to add nix to your shell.nix
2. **Getting Started > Create a makes project**
  The current documentation merely mentions the scenario of _Getting started as developer: Locate in the root of your project_. There's no mention of how to create such a project from scratch in the beginning. This section appears to be geared instead towards contributions to the makes project itself if I'm not mistaken. (Perhaps the section on migrating from existing nix projects also fits here).
3. **Some project starter examples**
  This would go a long way to enabling adoption of makes. e.g., creating a nodejs app (e.g., react app) and controlling the version of node required by the project. Obviously those familiar with nix can navigate the source to figure that out, but for those unfamiliar with nix, imho aren't going to spend days working through the nix pills if they can't generate a project and feel the success straight away.

**Comment by ldeck** [comment]
Just to add a further suggestion in terms of getting started:
The [aws cdk-workshop](https://cdkworkshop.com) provides a reasonably complete (albeit trivial) example of getting started with the cdk framework.

Given makes is presumably competing in that space, providing a similarly complete workshop — even showing how the same could be solved using makes, both locally for development and deploying to aws (and optionally other cloud providers), would go a long way to showing the viability of adopting makes.
